### ISSUE TITLE
masks not applied in composite_map

### ISSUE THREAD

**Issue by elastufka** [sunpy/sunpy#2490]
Masks are applied when using map.peek() or map.plot() for EUVI and COR (generic?) maps (first two images). Combining them into a composite map results in the mask being "forgotten" (final image).

```
def plot_with_mask(maplist,mask,palette):
''' Turn list of maps and corresponding masks arrays (currently length presumed to be 2) into a single composite map and plot'''
    scaled_maps=[]
    for m,mmask in zip(maplist,mask):
           scaled_maps.append(sunpy.map.Map(m.data, m.meta, mask=mmask.mask))
    scaled_map=align_maps(scaled_maps[0],scaled_maps[1])
    fig = plt.figure()
    scaled_map.plot(cmap=palette)
    fig.show()
```

```
def align_maps(m1,m2,a1=1,a2=1): 
''' Following example in Sunpy documentation'''
    comp_map=sunpy.map.Map(m1,m2,composite=True)
    comp_map.set_alpha(0,a1)
    comp_map.set_alpha(1,a2)
    return comp_map
```

![bug_wl](https://user-images.githubusercontent.com/11930844/36852582-4bce9c30-1d6c-11e8-843d-a5c30b6065d7.png)
![bug_uv](https://user-images.githubusercontent.com/11930844/36852583-4becd592-1d6c-11e8-87cf-a743af29eef2.png)
![bug_comp](https://user-images.githubusercontent.com/11930844/36852672-993196da-1d6c-11e8-8b4c-8bb2e6ab9e71.png)

**Comment by wafels** [comment]
I think the reason for this is that compositemap.plot does not look for the presence of a mask, but map.plot does.  Compare https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L1742-L1745 to 
https://github.com/sunpy/sunpy/blob/master/sunpy/map/compositemap.py#L434-L435

I believe that the fix is to look for the mask in compositemap just in the same way that map does.

**Comment by nabobalis** [comment]
Closed by #2492
